### PR TITLE
Enforce files in `src` have distinct names after lowercasing and are allowed on all OS's

### PR DESCRIPTION
**Pull request by ericphanson**
closes https://github.com/JuliaRegistries/RegistryCI.jl/issues/441

### CODE REVIEW

**Comment by ericphanson** [comment]
bors try

**Review by DilumAluthge — APPROVED** [review]

**Comment by DilumAluthge** [comment]
SGTM

File: src/AutoMerge/guidelines.jl [hunk 1]
```diff
@@ -549,6 +549,69 @@ const guideline_code_can_be_downloaded = Guideline(;
...
+
+function meets_file_dir_name_check(name)
+    # https://stackoverflow.com/a/31976060
+    idx = findfirst(n -> occursin(n, name), DISALLOWED_CHARS)
+    if idx !== nothing
+        return false, "contains character $(DISALLOWED_CHARS[idx]) which may not be valid as a file or directory name on some platforms"
+    end
+
+    base, ext = splitext(name)
+    if uppercase(name) in DISALLOWED_NAMES || uppercase(base) in DISALLOWED_NAMES
```
Comment (GunnarFarneback) [inline src/AutoMerge/guidelines.jl:579]: I believe the first term is redundant but it doesn't hurt for clarity.

**Review by GunnarFarneback — APPROVED** [review]

**Comment by ericphanson** [comment]
bors merge

**Pull request merged**